builtin/providers/aws/resource_aws_vpc_peering_connection.go

```diff
@@ -124,20 +124,38 @@ func resourceAwsVPCPeeringRead(d *schema.ResourceData, meta interface{}) error {
 			return nil
 		}
 	}
+	log.Printf("[DEBUG] VPC Peering Connection response: %#v", pc)
 
 	d.Set("accept_status", *pc.Status.Code)
-	d.Set("peer_owner_id", *pc.AccepterVpcInfo.OwnerId)
-	d.Set("peer_vpc_id", *pc.AccepterVpcInfo.VpcId)
-	d.Set("vpc_id", *pc.RequesterVpcInfo.VpcId)
 
-	err = d.Set("accepter", flattenPeeringOptions(pc.AccepterVpcInfo.PeeringOptions))
-	if err != nil {
-		return err
+	// When the VPC Peering Connection is pending acceptance,
+	// the details about accepter and/or requester peering
+	// options would not be included in the response.
+	if pc.AccepterVpcInfo != nil {
+		d.Set("peer_owner_id", *pc.AccepterVpcInfo.OwnerId)
+		d.Set("peer_vpc_id", *pc.AccepterVpcInfo.VpcId)
+
+		if _, ok := d.GetOk("accepter"); ok {
+			if pc.AccepterVpcInfo.PeeringOptions != nil {
+				err = d.Set("accepter", flattenPeeringOptions(pc.AccepterVpcInfo.PeeringOptions))
```

Comment:
It seems this will only read information on the `accepter` if the user has configured it locally, is that correct? That behavior seems counter to most of Terraform, in that it would prevent us from detecting configuration drift. We should read the `accepter` regardless, I think. Is there a reason for this behavior?